### ISSUE TITLE
Pagination Total count fails with Marten if sorting is present.

### ISSUE THREAD

**Issue by JacksonCribb** [ChilliCream/graphql-platform#6125]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Product

Hot Chocolate

### Describe the bug

Marten integration cannot currently provide a total count for pagination when any ordering is present.

This is due to pagination calling .Count() in which marten does not remove ordering statements leading to the following invalid sql:
```sql 
select count(*) as number from public.mt_doc_foo as d order by CAST(d.data ->> 'Bar' as integer)
```

### Steps to reproduce

1. Return an `IQueryable<T>` provided by Marten

```csharp
builder.Services
    .AddGraphQLServer()
    .AddProjections()
    .AddMartenFiltering()
    .AddMartenSorting()
    .AddQueryType<Query>();
```

```csharp 
[ObjectType("Query")]
public class Query
{
    [UsePaging(IncludeTotalCount = true)]
    [UseProjection]
    [UseFiltering]
    [UseSorting]
    public IMartenQueryable<Foo> GetFoos([Service] IQuerySession q) => q.Query<Foo>()
}
```

2. include any ordering statement with total count present in query  

2.1 via graphql:
```graphql
query {
  foos(order: {bar: ASC}) {
    totalCount
  }
}
```

2.2 apply ordering directly to an `IQueryable<T>`
```csharp 
[ObjectType("Query")]
public class Query
{
    [UsePaging(IncludeTotalCount = true)]
    [UseProjection]
    [UseFiltering]
    [UseSorting]
    public IQueryable<Foo> GetFoos([Service] IQuerySession q) => q.Query<Foo>().OrderBy(d => d.Bar)
}
```

### Relevant log output

```shell
Marten Command Failure:$\nselect count(*) as number from public.mt_doc_foo as d order by CAST(d.data ->> 'Bar' as integer)$\n$\n42803: column \"d.data\" must appear in the GROUP BY clause or be used in an aggregate function\n\nPOSITION: 69
  at JasperFx.Core.Exceptions.ExceptionTransformExtensions.TransformAndThrow(IEnumerable`1 transforms, Exception ex)
   at JasperFx.Core.Exceptions.ExceptionTransforms.TransformAndThrow(Exception ex)
   at Marten.Exceptions.MartenExceptionTransformer.WrapAndThrow(NpgsqlCommand command, Exception exception)
   at Marten.Internal.Sessions.QuerySession.handleCommandException(NpgsqlCommand cmd, Exception e)
   at Marten.Internal.Sessions.QuerySession.ExecuteReader(NpgsqlCommand command)
   at Marten.Linq.MartenLinqQueryProvider.ExecuteHandler[T](IQueryHandler`1 handler)
--- End of stack trace from previous location ---
   at JasperFx.Core.Exceptions.ExceptionTransformExtensions.TransformAndThrow(IEnumerable`1 transforms, Exception ex)
   at JasperFx.Core.Exceptions.ExceptionTransforms.TransformAndThrow(Exception ex)
   at Marten.Exceptions.MartenExceptionTransformer.WrapAndThrow(Exception exception)
   at Marten.Linq.MartenLinqQueryProvider.ExecuteHandler[T](IQueryHandler`1 handler)
   at Marten.Linq.MartenLinqQueryProvider.Execute[TResult](Expression expression)
   at HotChocolate.Types.Pagination.QueryableCursorPagingHandler`1.ResolveAsync(IResolverContext context, IQueryable`1 source, CursorPagingArguments arguments, CancellationToken cancellationToken)
   at HotChocolate.Types.Pagination.CursorPagingHandler.HotChocolate.Types.Pagination.IPagingHandler.SliceAsync(IResolverContext context, Object source)
   at HotChocolate.Types.Pagination.PagingMiddleware.InvokeAsync(IMiddlewareContext context)
   at HotChocolate.Utilities.MiddlewareCompiler`1.ExpressionHelper.AwaitTaskHelper(Task task)
   at HotChocolate.Execution.Processing.Tasks.ResolverTask.ExecuteResolverPipelineAsync(CancellationToken cancellationToken)
   at HotChocolate.Execution.Processing.Tasks.ResolverTask.TryExecuteAsync(CancellationToken cancellationToken)
```

### Version

13.0.5

**Comment by JacksonCribb** [comment]
I've opened a companion issue on the Marten side too as this may be entirely resolvable upstream.

**Comment by kamalsivalingam** [comment]
We are running into the same error as well. Is there any interim fix?

**Comment by glen-84** [comment]
@JacksonCribb Is this issue resolved with Marten 7.0.0-beta.2?
